frontend/packages/multicluster-sdk/src/components/FleetResourceEventStream/EventStream.tsx

```diff
@@ -0,0 +1,203 @@
+
+              return sortEvents([...topEvents, object])
+            case 'DELETED':
+              return topEvents.filter((e) => e?.metadata?.uid !== uid)
+            default:
+              // eslint-disable-next-line no-console
+              console.error(`UNHANDLED EVENT: ${eventType}`)
+              return topEvents
+          }
+        })
+      }
+
+      ws.current.onopen = () => {
+        setActive(true)
+        setError(false)
+        setLoading(false)
+      }
+
+      ws.current.onclose = (evt: CloseEvent) => {
+        setActive(false)
+        if (evt?.wasClean === false) {
+          setError(evt.reason || t('public~Connection did not close cleanly.'))
+        }
+      }
+
+      ws.current.onerror = () => {
+        setActive(false)
+        setError(true)
+      }
+    }
+
+    return () => {
+      ws.current?.close()
+    }
+  }, [namespace, fieldSelector, active, t, resource.cluster, hubCluster, loaded, backendAPIPath])
+
+  const count = sortedEvents.length
+  const noEvents = count === 0
+  const noMatches = count > 0 && count === 0
+  let sysEventStatus, statusBtnTxt
+
+  if (noEvents || (noMatches && resource)) {
+    sysEventStatus = <EmptyState title={t('public~No events')}></EmptyState>
+  }
+  if (noMatches && !resource) {
+    sysEventStatus = (
+      <EmptyState title={t('public~No matching events')}>
+        {count >= MAX_MESSAGES
+          ? t('public~{{count}}+ event exist, but none match the current filter', {
+              count: MAX_MESSAGES,
+            })
+          : t('public~{{count}} event exist, but none match the current filter', {
+              count,
+            })}
+      </EmptyState>
+    )
+  }
+
+  if (error) {
+    statusBtnTxt = (
+      <span className="co-sysevent-stream__connection-error">
+        {_.isString(error)
+          ? t('public~Error connecting to event stream: { error }', {
+              error,
+            })
+          : t('public~Error connecting to event stream')}
+      </span>
+    )
+    sysEventStatus = (
+      <EmptyState title={t('public~Error loading events')}>
+        {t('public~An error occurred during event retrieval. Attempting to reconnect...')}
+      </EmptyState>
+    )
+  } else if (loading) {
+    statusBtnTxt = <span>{t('public~Loading events...')}</span>
+    sysEventStatus = <Spinner />
+  } else if (active) {
+    statusBtnTxt = <span>{t('public~Streaming events...')}</span>
+  } else {
+    statusBtnTxt = <span>{t('public~Event stream is paused.')}</span>
+  }
+
+  const klass = css('co-sysevent-stream__timeline', {
+    'co-sysevent-stream__timeline--empty': !count,
+  })
+  const messageCount =
+    count < MAX_MESSAGES
+      ? t('public~Showing {{count}} event', { count })
+      : t('public~Showing most recent {{count}} event', { count })
+
+  const toggleStream = () => {
+    setActive((prev) => !prev)
+  }
+
+  if (!resource.cluster || resource.cluster === hubCluster) return <ResourceEventStream resource={resource} />
```

Comment:
Am I wrong to think that this will give a stack overflow? When cluster equals the `hubCluster` it calls itself with the same resource so that doesn't change the logic and it will call itself again